src/AssetsPage/AssetsTable/AssetsTable.test.jsx

```diff
@@ -24,31 +28,360 @@ const defaultProps = {
       display_name: 'bird.json',
       id: 'bird.json',
     },
+    {
+      display_name: 'fish.doc',
+      id: 'fish.doc',
+    },
   ],
   assetsParameters: {
-    assetTypes: {
-      courseId: 'test-course',
-      page: '0',
-      pageSize: '50',
-      sort: 'sort',
-    },
+    courseId: 'test-course',
+    page: '0',
+    pageSize: '50',
   },
   assetsStatus: {},
-  deleteAsset: () => {},
   clearAssetsStatus: () => {},
+  deleteAsset: () => {},
+  updateSort: () => {},
 };
 
-let wrapper;
-
 describe('<AssetsTable />', () => {
-  it('renders correct number of assets', () => {
-    wrapper = mount(
-      <AssetsTable
-        {...defaultProps}
-      />,
-    );
-    // + 1 because it includes row item for headings
-    expect(wrapper.find('tr')).toHaveLength(defaultProps.assetsList.length + 1);
+  describe('renders', () => {
+    beforeEach(() => {
+      wrapper = mount(
+        <AssetsTable
+          {...defaultProps}
+        />,
+      );
+    });
+    it(' correct number of assets', () => {
+      // + 1 because it includes row item for headings
+      expect(wrapper.find('tr')).toHaveLength(defaultProps.assetsList.length + 1);
+    });
+    it('correct number of trashcan buttons', () => {
+      expect(wrapper.find('button').filterWhere(button => button.hasClass('fa-trash'))).toHaveLength(defaultProps.assetsList.length);
+    });
+  });
+  describe('onSortClick', () => {
+    let updateSortSpy;
+
+    beforeEach(() => {
+      updateSortSpy = jest.fn();
+    });
+
+    const sortableAssetProps = [
+      {
+        sort: 'date_added',
+        direction: 'desc',
+      },
+      {
+        sort: 'date_added',
+        direction: 'asc',
+      },
+      {
+        sort: 'date_added',
+        direction: 'desc',
+      },
+    ];
+
+    const assertValues = [
+      {
+        buttonIndexToClick: 2,
+        buttonToClickText: 'Date Added',
+        onSortColumnParameter: 'date_added',
+        onSortDirectionParameter: 'asc',
+      },
+      {
+        buttonIndexToClick: 2,
+        buttonToClickText: 'Date Added',
+        onSortColumnParameter: 'date_added',
+        onSortDirectionParameter: 'desc',
+      },
+      {
+        buttonIndexToClick: 0,
+        buttonToClickText: 'Name',
+        onSortColumnParameter: 'display_name',
+        onSortDirectionParameter: 'desc',
+      },
+    ];
+
+    sortableAssetProps.forEach((sortableAssetProp, index) => {
```

Comment:
For each sortableAssetProp, we dynamically generate a test that starts with a sort state that's specified by sortableAssetProp.sort and sortableAssetProp.direction.